### PR TITLE
copr: (refactor) Lift heap struct out from top_n_executor

### PR DESCRIPTION
**Pull request by LittleFall**
### What is changed and how it works?

Issue Number: ref https://github.com/tikv/tikv/issues/13936

What's Changed: Lift out heap struct from top_n_executor to convenient operator PartitionTopN's implementation.

### Check List 

Tests 

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by LittleFall** [comment]
unit test failed

<details><summary>tests::integrations raftstore::test_replication_mode::test_switching_replication_mode_hibernate</summary>
<p>

```
[2023-01-30T07:47:16.420Z] ------------

[2023-01-30T07:47:16.420Z]      Summary [ 202.473s] 1569 tests run: 1568 passed (4 flaky), 1 failed, 1624 skipped

[2023-01-30T07:47:16.420Z]    FLAKY 2/3 [   3.489s]                               raftstore store::worker::region::tests::test_pending_applies

[2023-01-30T07:47:16.420Z]    FLAKY 2/3 [   0.548s]                               raftstore store::worker::region::tests::test_stale_peer

[2023-01-30T07:47:16.420Z]    FLAKY 2/3 [   1.283s]                       tests::failpoints cases::test_replica_read::test_read_index_after_transfer_leader

[2023-01-30T07:47:16.420Z]    FLAKY 2/3 [   4.175s]                     tests::integrations raftstore::test_snap_recovery::test_snap_wait_apply

[2023-01-30T07:47:16.420Z]   TRY 3 FAIL [   2.282s]                     tests::integrations raftstore::test_replication_mode::test_switching_replication_mode_hibernate

[2023-01-30T07:47:16.420Z] 

[2023-01-30T07:47:16.420Z] --- TRY 3 STDOUT:                            tests::integrations raftstore::test_replication_mode::test_switching_replication_mode_hibernate ---

[2023-01-30T07:47:16.420Z] 

[2023-01-30T07:47:16.420Z] running 1 test

[2023-01-30T07:47:16.420Z] test raftstore::test_replication_mode::test_switching_replication_mode_hibernate ... FAILED

[2023-01-30T07:47:16.420Z] 

[2023-01-30T07:47:16.420Z] failures:

[2023-01-30T07:47:16.420Z] 

[2023-01-30T07:47:16.420Z] failures:

[2023-01-30T07:47:16.420Z]     raftstore::test_replication_mode::test_switching_replication_mode_hibernate

[2023-01-30T07:47:16.420Z] 

[2023-01-30T07:47:16.420Z] test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 540 filtered out; finished in 2.16s

[2023-01-30T07:47:16.420Z] 

[2023-01-30T07:47:16.420Z] 

[2023-01-30T07:47:16.420Z] --- TRY 3 STDERR:                            tests::integrations raftstore::test_replication_mode::test_switching_replication_mode_hibernate ---

[2023-01-30T07:47:16.420Z] thread 'raftstore::test_replication_mode::test_switching_replication_mode_hibernate' panicked at 'assertion failed: `(left == right)`

[2023-01-30T07:47:16.420Z]   left: `SimpleMajority`,

[2023-01-30T07:47:16.420Z]  right: `IntegrityOverLabel`', tests/integrations/raftstore/test_replication_mode.rs:392:5

[2023-01-30T07:47:16.420Z] stack backtrace:

[2023-01-30T07:47:16.420Z]    0: rust_begin_unwind

[2023-01-30T07:47:16.420Z]              at /rustc/96ddd32c4bfb1d78f0cd03eb068b1710a8cebeef/library/std/src/panicking.rs:575:5

[2023-01-30T07:47:16.420Z]    1: core::panicking::panic_fmt

[2023-01-30T07:47:16.420Z]              at /rustc/96ddd32c4bfb1d78f0cd03eb068b1710a8cebeef/library/core/src/panicking.rs:65:14

[2023-01-30T07:47:16.420Z]    2: core::panicking::assert_failed_inner

[2023-01-30T07:47:16.420Z]    3: core::panicking::assert_failed

[2023-01-30T07:47:16.420Z]              at /rustc/96ddd32c4bfb1d78f0cd03eb068b1710a8cebeef/library/core/src/panicking.rs:201:5

[2023-01-30T07:47:16.420Z]    4: integrations::raftstore::test_replication_mode::test_switching_replication_mode_hibernate

[2023-01-30T07:47:16.420Z]              at /home/jenkins/tikv-src/tests/integrations/raftstore/test_replication_mode.rs:392:5

[2023-01-30T07:47:16.420Z]    5: integrations::raftstore::test_replication_mode::test_switching_replication_mode_hibernate::{{closure}}

[2023-01-30T07:47:16.420Z]              at /home/jenkins/tikv-src/tests/integrations/raftstore/test_replication_mode.rs:354:48

[2023-01-30T07:47:16.420Z]    6: core::ops::function::FnOnce::call_once

[2023-01-30T07:47:16.420Z]              at /rustc/96ddd32c4bfb1d78f0cd03eb068b1710a8cebeef/library/core/src/ops/function.rs:513:5

[2023-01-30T07:47:16.420Z]    7: test_util::runner::run_test_with_hook::{{closure}}::{{closure}}

[2023-01-30T07:47:16.420Z]    8: core::ops::function::FnOnce::call_once{{vtable.shim}}

[2023-01-30T07:47:16.420Z]    9: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once

[2023-01-30T07:47:16.420Z]              at /rustc/96ddd32c4bfb1d78f0cd03eb068b1710a8cebeef/library/alloc/src/boxed.rs:2000:9

[2023-01-30T07:47:16.420Z] note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

[2023-01-30T07:47:16.420Z] 

[2023-01-30T07:47:16.420Z] error: test run failed

[2023-01-30T07:47:16.420Z] + gdb -c 'core.*' -batch -ex 'info threads' -ex 'thread apply all bt'

[2023-01-30T07:47:16.420Z] /home/jenkins/agent/tikv-master/build/core.*: No such file or directory.

[2023-01-30T07:47:16.420Z] No threads.

[2023-01-30T07:47:16.420Z] + exit 1
```


</p>
</details>

**Review by gengliqi — APPROVED** [review]

File: components/tidb_query_executors/src/util/mod.rs [hunk 1]
```diff
@@ -1,15 +1,16 @@
 // Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
 
+use tidb_query_common::Result;
```
Comment (sticnarf) [inline components/tidb_query_executors/src/util/mod.rs:3]: Unnecessary moves. And it's more common to put `use` below `mod` (though it's not mandatory in TiKV now)

**Review by sticnarf — APPROVED** [review]
Mostly LGTM

File: components/tidb_query_executors/src/util/mod.rs (hunk 1 above)
Comment (LittleFall) [reply components/tidb_query_executors/src/util/mod.rs]: addressed. It was auto-formatted by my Clion. I have disabled it and replaced it with rustfmt. 😈

**Comment by LittleFall** [comment]
test failed https://ci.pingcap.net/blue/organizations/jenkins/tikv_ghpr_test/detail/tikv_ghpr_test/31169/tests
```
test::async_commit – backup-stream::integration
16s
cases::test_pd_client::test_retry – tests::failpoints
2s
raftstore::test_witness::test_witness_replica_read – tests::integrations
4s
test::async_commit – backup-stream::integration
16s
cases::test_pd_client::test_retry – tests::failpoints
2s
raftstore::test_witness::test_witness_replica_read – tests::integrations
4s
```

**Pull request merged**